src/validation.h

```diff
@@ -505,6 +505,41 @@ class BlockManager {
         CBlockIndex** ppindex) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
 };
 
+/**
+ * A convenience class for constructing the CCoinsView* hierarchy used
+ * to facilitate access to the UTXO set.
+ *
+ * This class consists of an arrangement of layered CCoinsView objects,
+ * preferring to store and retrieve coins in memory via `m_cacheview` but
+ * ultimately falling back on cache misses to the canonical store of UTXOs on
+ * disk, `m_dbview`.
+ */
+class CoinsViews {
+
+public:
+    //! The lowest level of the CoinsViews cache hierarchy sits in a leveldb database on disk.
+    //! All unspent coins reside in this store.
+    std::unique_ptr<CCoinsViewDB> m_dbview GUARDED_BY(cs_main);
+
+    //! This view wraps access to the leveldb instance and handles read errors gracefully.
+    std::unique_ptr<CCoinsViewErrorCatcher> m_catcherview GUARDED_BY(cs_main);
```

Comment:
In commit 087bdf7190 refactor: have CCoins* data managed under CChainState


Why is this a pointer?